module-new/src/components/checkbox/checkbox.component.tsx

```diff
@@ -0,0 +1,171 @@
+  scrollValidationErrorsIntoView?: boolean;
+}
+
+export const Checkbox = React.forwardRef(
+  <TData extends BindType>(
+    {
+      bind,
+      checkboxClassName,
+      checked,
+      customIndicator,
+      customIndeterminateIndicator,
+      direction,
+      disabled,
+      onCheckedChange,
+      label,
+      labelClassName,
+      scrollValidationErrorsIntoView,
+      testId,
+      validationErrorsClassName,
+      validationErrorMessages,
+      ...nativeProps
+    }: ICheckboxProps<TData>,
+    ref: React.ForwardedRef<HTMLButtonElement>
+  ) => {
+    const id = React.useId();
+
+    const globals = useArmstrongConfig({
+      scrollValidationErrorsIntoView,
+      checkboxCustomIndicator: customIndicator,
+      checkboxCustomIndeterminateIndicator: customIndeterminateIndicator,
+    });
+
+    const [boundValue, setBoundValue, bindConfig] = useBindingState(bind, {
+      value: checked,
+      onChange: onCheckedChange,
+      validationErrorMessages,
+      validationMode: globals.validationMode,
+    });
+
+    const onCheckedChangeInternal = React.useCallback<Required<CheckboxProps>['onCheckedChange']>(
+      newValue => {
+        setBoundValue?.(newValue as TData);
+      },
+      [setBoundValue]
+    );
+
+    const indicator = React.useMemo(() => {
+      switch (boundValue) {
+        case 'indeterminate':
+          return globals.checkboxCustomIndeterminateIndicator;
+        default:
+          return globals.checkboxCustomIndicator;
+      }
+    }, [boundValue, globals.checkboxCustomIndicator, globals.checkboxCustomIndeterminateIndicator]);
+
+    return (
+      <StatusWrapper
+        className={concat(validationErrorsClassName, 'arm-input-base')}
+        validationMode={bindConfig.validationMode}
+        errorIcon={bindConfig.validationErrorIcon}
+      >
+        <div
+          className={concat('arm-checkbox-container', checkboxClassName)}
+          data-disabled={disabled}
+          data-direction={direction}
+          data-testid={testId}
+          {...nativeProps}
+        >
+          <Root
+            className="arm-checkbox"
+            disabled={disabled}
+            id={id}
+            checked={boundValue ?? undefined}
+            onCheckedChange={onCheckedChangeInternal}
+            ref={ref}
+          >
+            <Indicator className="arm-checkbox-indicator" asChild>
+              {indicator}
+            </Indicator>
+          </Root>
+
+          {label && (
+            <Label
+              className={concat('arm-checkbox-label', labelClassName)}
+              data-direction={direction}
+              data-disabled={disabled}
+              htmlFor={id}
+            >
+              {label}
+            </Label>
+          )}
+        </div>
+        {!!bindConfig.validationErrorMessages?.length && bindConfig.shouldShowValidationErrorMessage && (
+          <ValidationErrors
+            className={validationErrorsClassName}
+            validationMode={'message'}
```

Comment:
Hmmmm, seems a bit opinionated 😂

Reply:
That's just your _opinion_. (Changed in update)